framework/src/actioncomponents/CylinderComponent.C

```diff
@@ -38,9 +38,20 @@ CylinderComponent::validParams()
   params.addRequiredRangeCheckedParam<Real>("radius", "radius>0", "Radius of the cylinder");
   params.addRequiredRangeCheckedParam<Real>("length", "length>0", "Length/Height of the cylinder");
 
+  params.addParam<unsigned int>("n_radial",
+                                "Number of radial elements. Only assign if mesh is 2D!");
+  params.addParam<unsigned int>(
+      "n_radial_rings",
+      1,
+      "Number of radial rings within the mesh (does not affect boundary layers).");
+
   params.addRequiredParam<unsigned int>("n_axial", "Number of axial elements of the cylinder");
-  params.addParam<unsigned int>("n_radial", "Number of radial elements of the cylinder");
-  params.addParam<unsigned int>("n_azimuthal", "Number of azimuthal elements of the cylinder");
+  params.addRangeCheckedParam<Real>("boundary_layer_width",
```

Comment:
Are boundary layers applied to both axial and radial boundaries?

Reply:
Just an idea, which you can decline: consider instead of hard-wiring this "boundary layer" just giving the user total freedom of radial regions and numbers of elements in each.